contracts/ServiceNodeContribution.sol

```diff
@@ -184,20 +196,55 @@ contract ServiceNodeContribution is Shared {
     function cancelNode() public onlyOperator {
         require(!finalized, "Cannot cancel a finalized node.");
         require(!cancelled, "Node has already been cancelled.");
+        removeAndRefundContributor(msg.sender);
+        cancelled = true;
```

Comment:
https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html

this should be above the transfer

Reply:
I have fixed this and put the `cancelled` above the function call.